safehttp/mux.go

```diff
@@ -164,11 +182,28 @@ func (m methodHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	for _, intercep := range m.muxInterceps {
-		if res := intercep.Before(rw, ir); res.written {
+	h.ServeHTTP(w, r)
+}
+
+// handlerWithInterceptors encapsulates a handler and its corresponding
+// interceptors.
+type handlerWithInterceptors struct {
+	handler   Handler
+	interceps map[string]Interceptor
+	disp      Dispatcher
+}
+
+// ServeHTTP calls the Before method of all the interceptors and then calls the
+// underlying handler.
+func (h handlerWithInterceptors) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	rw := NewResponseWriter(h.disp, w, h.interceps)
+	ir := NewIncomingRequest(r)
+
+	for _, it := range h.interceps {
```

Comment:
This line makes it nondeterministic (I actually have missed this issue before, as it was a `map` earlier too).

Reply:
We now have a slice of interceptor keys to iterate over instead.